Demo/src/main/java/com/braintreepayments/demo/DemoActivity.java

```diff
@@ -51,15 +45,13 @@ protected void onCreate(Bundle savedInstanceState) {
         requestWindowFeature(Window.FEATURE_INDETERMINATE_PROGRESS);
         setContentView(R.layout.activity_demo);
 
-        authProvider = new DemoAuthorizationProvider();
-        if (savedInstanceState != null) {
-            authorization = savedInstanceState.getString(EXTRA_AUTHORIZATION);
-            customerId = savedInstanceState.getString(EXTRA_CUSTOMER_ID);
-        }
-
         setupActionBar();
         setProgressBarIndeterminateVisibility(true);
 
+        // perform reset on preference change
+        Settings.getPreferences(this)
+                .registerOnSharedPreferenceChangeListener((sharedPreferences, s) -> performReset());
```

Comment:
This is a bit of a hack but we should be able to remove it once each fragment owns the creation of its own `BraintreeClient`. In the future, the concept of a "reset" may no longer be necessary in the Demo app.